### PR TITLE
Feature/ak playable game board

### PR DESCRIPTION
**Pull request by annamkummer**
# What (if any) features are you implementing?
- User can select (stamp) grid squares
- When user clicks "BINGO!" button, state will update to reflect whether the board has a win or not
# What (if anything) did you refactor?
- Moved some functions from Game to utils.js
# Were there any issues that arose?
- There is an infinite loop happening when all countries are called for a region without the user clicking BINGO. We need to determine how to handle that scenario - do they lose, keep getting coordinates?
# Is there anything that you need from your teammate?
- No
# Any other comments, questions, or concerns?
-

### CODE REVIEW

File: src/components/Game.js [hunk 1]
```diff
@@ -1,37 +1,200 @@
...
+    if (latNum < 0) {
+      latNum = latNum * -1
+      latDir = 'S'
+    }
+    if (longNum < 0) {
+      longNum = longNum * -1
+      longDir = 'W'
+    }
+
+    console.log('called country: ', randomCountry.name)
```
Comment (NEwertKrocker) [inline src/components/Game.js:47]: Oh man, what a useful bit of functionality for testing. Thanks for including this.

File: src/components/Game.js [hunk 2]
```diff
@@ -1,37 +1,200 @@
...
+    }
+
+    let squareCountries = randomCountries.map((randomCountry, index) => {
+      return {
+        name: randomCountry.name,
+        id: randomCountry.id,
+        flag_path: randomCountry.flag_path,
+        space: index,
+        isStamped: false,
+        correct: null
```
Comment (NEwertKrocker) [inline src/components/Game.js:104]: I assume this `correct` property is there for the purposes of win verification.

File: src/components/Game.js [hunk 3]
```diff
@@ -1,37 +1,200 @@
...
+      let errors = 0
+        wins.forEach(win => {
+          win.forEach(square => {
+            if (this.state.calledCoordinateIds.includes(this.state.currentBoard[square].id)) {
+              this.setState(prevState => {
+                return prevState.currentBoard[square].correct = true
+              })
+            } else {
+              errors += 1
+              this.setState({winStatus: 'Oops! Not all of those countries were called - check the board for details'})
```
Comment (NEwertKrocker) [inline src/components/Game.js:142]: Does this account for both error possibilities -- that the user incorrectly selected uncalled squares AND that they may have neglected to select squares that were called? (And do we care?)

File: src/components/GameBoard.js [hunk 4]
```diff
@@ -1,33 +1,23 @@
...
-        "Iowa",
-        "Arkansas",
-        "Missouri",
-        "Louisiana",
-        "Michigan",
-        "Indiana",
-        "Ohio",
-        "Illinois",
-        "Pennsylvania"
-    ]
```
Comment (NEwertKrocker) [inline src/components/GameBoard.js:22]: Get outta here, fake countries! Come back when you declare independence!

File: src/utils.js [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+    [8, 9, 10, 11],
+    [12, 13, 14, 15],
+// Down:
+    [0, 4, 8, 12],
+    [1, 5, 9, 13],
+    [2, 6, 10, 14],
+    [3, 7, 11, 15],
+// Diag:
+    [0, 5, 10, 15],
+    [3, 6, 9, 12],
```
Comment (NEwertKrocker) [inline src/utils.js:14]: Excellent work here. So neatly laid out!

**Review by NEwertKrocker — APPROVED** [review]
This is excellent! I pulled the branch down and played with it and it works like a dream! 

That said, now that I have some experience as a "player," I wonder if asking the user to guess so many things correctly before giving any feedback isn't a little harsh. (I was cavalier in my earlier assessments, I think.) Do you think we should try to implement some move-by-move feedback? Tell users they made a mistake AS they're making it? Let's talk about it when next we meet.

**Review by jkiernan12 — APPROVED** [review]
Amazing work on this!!! I noticed a few bugs, but we can talk more about it during our meeting. Again, awesome work!

**Pull request merged**